src/eslint_result.ts

```diff
@@ -0,0 +1,98 @@
+import * as core from "@actions/core"
+import { ChangeRange, generateChangeRanges } from "./git_utils"
+
+export type ResultObject = {
+  filePath: string
+  messages: EslintMessage[]
+}
+
+export type EslintResultInstance = InstanceType<typeof EslintResult>
+
+type EslintMessage = {
+  ruleId: string
+  severity: number
+  message: string
+  line: number
+  column: number
+  nodeType: string
+  messageId: string
+  endLine: number
+  endColumn: number
+  suggestions: any[]
+}
+
+export class EslintResult {
+  public relevantWarningCount: number = 0
+  public relevantErrorCount: number = 0
+  private resultObject: ResultObject
+  private compareSha: string
+  private changeRanges?: ChangeRange[]
+  private relevantMessages: EslintMessage[] = []
+
+  constructor(resultObject: ResultObject, compareSha: string) {
+    this.resultObject = resultObject
+    this.compareSha = compareSha
+  }
+
+  async asyncInitialize() {
+    this.changeRanges = await generateChangeRanges(
+      this.resultObject.filePath,
```

Comment:
Is `filePath` relative or absolute? My headspace for the question comes from noting distinction in how we build the array of changed files. Sometimes those are relative to the `workingDir` vs relative to the git root. However, based on my checking locally, `git diff --name-only` always outputs the path from the git root

Reply:
That is a good question. After looking myself, it seems `filePath` from eslint is an absolute path. Git appears to not mind that we are calling diff on an absolute path.

The file paths that _we_ give to eslint are from the working directory though, as we manipulate the path to scope it to the working directory here: https://github.com/planningcenter/balto-eslint/blob/2f523c73941b74566ba8951a46e991e99f98f055/src/git_utils.ts#L53-L55

Reply:
Nice. Ok, that makes sense. Absolute file paths _from_ `eslint` seem good, and it makes sense that `git` would be happy with that. Thanks for answering!